src/categories/categories.controller.ts

```diff
@@ -1,34 +1,40 @@
 import { Controller, Get, Post, Body, Patch, Param, Delete } from '@nestjs/common';
+import { ApiOperation } from '@nestjs/swagger';
+import { ZaLaResponse } from '../common/helpers/response';
 import { CategoriesService } from './categories.service';
 import { CreateCategoryDto } from './dto/create-category.dto';
 import { UpdateCategoryDto } from './dto/update-category.dto';
 
 @Controller('categories')
 export class CategoriesController {
-  constructor(private readonly categoriesService: CategoriesService) {}
+  constructor(private readonly categoryService: CategoriesService) {}
 
-  @Post()
-  create(@Body() createCategoryDto: CreateCategoryDto) {
-    return this.categoriesService.create(createCategoryDto);
+  @Post('/create')
+  @ApiOperation({summary: 'Create a new category'})
+  async create(@Body() createCategoryDto: CreateCategoryDto) {
```

Comment:
What is this async function returning? a promise, `Promise<Ok<Category>>` A lot of codes in the dummy core can give you hints, study the way they were written

Also replicate it in other controllers and service files

Reply:
Noted... I will return the types properly

Reply:
I was able to fix a minor bug... Thanks for this